cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisioningService.java

```diff
@@ -51,19 +54,24 @@
    */
   public void reloadProvisioners() {
     Map<String, Provisioner> provisioners = provisionerProvider.loadProvisioners();
+    Map<String, ProvisionerConfig> provisionerConfigs = provisionerConfigProvider.loadProvisionerConfigs();
     LOG.debug("Provisioners = {}", provisioners);
-    Map<String, ProvisionerSpecification> specs = new HashMap<>(provisioners.size());
+    Map<String, ProvisionerDetail> details = new HashMap<>(provisioners.size());
     for (Map.Entry<String, Provisioner> provisionerEntry : provisioners.entrySet()) {
-      specs.put(provisionerEntry.getKey(), provisionerEntry.getValue().getSpec());
+      ProvisionerSpecification spec = provisionerEntry.getValue().getSpec();
+      String provisionerName = provisionerEntry.getKey();
+      details.put(provisionerName,
+                  new ProvisionerDetail(spec.getName(), spec.getDescription(),
+                                        provisionerConfigs.get(provisionerName).getConfigurationGroups()));
```

Comment:
provisionerConfigs is not guaranteed to have this file. If it doesn't exist, use an empty List